### DISCUSSION TITLE
No "Alternative Run" available after building for Ubuntu 20.04

### DISCUSSION THREAD

**Discussion by rafaelmgallego** [Audiveris/audiveris#472] [Q&A] [answered]
I have built Audiveris following the directions in https://bacchushlg.gitbooks.io/audiveris-5-1/content/install/sources.html
To run again the program without rebuilding there is the "Alternative Run" option extracting the files Audiveris.zio o .tar under the folder built/distributions,but thos folder does not appear in the folder I have built the code (nor anywhere that I could find).
Is there something else I should do that it is not in the mentioned directions page?
I will appreciate your help.

**Comment by maximumspatium** [comment]
Hi Rafael, does your Audiveris build actually run? You can try it out with `./gradlew run` command...

**Comment by rafaelmgallego** [comment]
Yes, I can run it with "./gradlew run".

The question is that in order to build and run it I have to change to 
Java 8, and I was wondering if after building the code, I could go back 
to Java 11 and run it through the "alternative run" option. But the 
build/distributions folder has not been created.

Thanks a lot for your prompt answer!

On 16/3/21 10:40, Maxim Poliakovski wrote:
>
> Hi Rafael, does your Audiveris build actually run? You can try it out 
> with |./gradlew run| command...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Audiveris/audiveris/discussions/472#discussioncomment-487743>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AF5G6ISMUVF6HL2LAIX5LHLTD4RSHANCNFSM4ZHGNQQA>.
>
-- 
------------------------------------------------------------------------
RAFAEL GALLEGO SEVILLA
Departamento de Mecànica de Estructuras e Ingenierìa Hidràulica
Correo-e 	***@***.*** ***@***.***>
Teléfono 	+34 958 24 89 55 <tel:+34 958 24 89 55>
Sede 	ETS Ingeniería de Caminos, Canales y Puertos 
<https://goo.gl/maps/55ZTWqr2w7yFj7qU8>
Universidad de Granada, Avd. Fuentenueva s/n, Granada, 18071, España 
[Despacho 5, plt 4ª]
Página Web 	http://sl.ugr.es/gallego <http://sl.ugr.es/gallego>

Este mensaje se dirige exclusivamente a su destinatario y puede contener 
información privilegiada o confidencial. Si no es Ud. el destinatario 
indicado, queda notificado de que la utilización, divulgación o copia 
sin autorización está prohibida en virtud de la legislación vigente. 
Si ha recibido este mensaje por error, se ruega lo comunique 
inmediatamente por esta misma vía y proceda a su destrucción.

This message is intended exclusively for its addressee and may contain 
information that is CONFIDENTIAL and protected by professional 
privilege. If you are not the intended recipient you are hereby notified 
that any dissemination, copy or disclosure of this communication is 
strictly prohibited by law. If this message has been received in error, 
please immediately notify us via e-mail.

**Reply by maximumspatium** [reply]
Please run the following sequence of terminal commands from the folder you've cloned Audiveris' source into:

```
java -version
./gradlew clean build
ls build
```

The very first command prints out Java version being currently active. What does it say?
The 2nd command will initiate a clean build of Audiveris. Please ensure it succeeds (indicated by the **BUILD SUCCESSFUL in Xm Xs message** at the end)
The last command lists the content of the build directory. In the case of a successful build, you'll see the distribution directory there.

**Comment by rafaelmgallego** [comment, marked as answer]
I changed first to Java 8 and then issued the commands you told me: it 
created succesfully the build/distributions folder with the ZIP and TAR 
files.

Thanks a lot for your help!.

Nevertheless, it appears I need to keep Java 8 in order to run the built 
code through "java -cp 
"/home/rgs/.audiveris/audiveris/build/distributions/Audiveris/lib/*" 
Audiveris", something it might be obvious for you, but not for me since 
I am fairly ignorant about these issues. It is not a problem, but I had 
a lot of trouble to run other java programs in the past, depending on 
the java version, and I do not want them to crash if I downgrade to java 
8. I'll modify your  bash file to change to java 8 prior to running 
audiveris.

Now I'll study th Manual to see if I amable to OMR my sheets.

Best regards, and thanks again.


On 16/3/21 11:38, Maxim Poliakovski wrote:
>
> Please run the following sequence of terminal commands from the folder 
> you've cloned Audiveris' source into:
>
> |java -version ./gradlew clean build ls build |
>
> The very first command prints out Java version being currently active. 
> What does it say?
> The 2nd command will initiate a clean build of Audiveris. Please 
> ensure it succeeds (indicated by the *BUILD SUCCESSFUL in Xm Xs 
> message* at the end)
> The last command lists the content of the build directory. In the case 
> of a successful build, you'll see the distribution directory there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/Audiveris/audiveris/discussions/472#discussioncomment-487977>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AF5G6IUDZUQLTGFAQLJMHLTTD4YKLANCNFSM4ZHGNQQA>.
>
-- 
------------------------------------------------------------------------
RAFAEL GALLEGO SEVILLA
Departamento de Mecànica de Estructuras e Ingenierìa Hidràulica
Correo-e 	***@***.*** ***@***.***>
Teléfono 	+34 958 24 89 55 <tel:+34 958 24 89 55>
Sede 	ETS Ingeniería de Caminos, Canales y Puertos 
<https://goo.gl/maps/55ZTWqr2w7yFj7qU8>
Universidad de Granada, Avd. Fuentenueva s/n, Granada, 18071, España 
[Despacho 5, plt 4ª]
Página Web 	http://sl.ugr.es/gallego <http://sl.ugr.es/gallego>

Este mensaje se dirige exclusivamente a su destinatario y puede contener 
información privilegiada o confidencial. Si no es Ud. el destinatario 
indicado, queda notificado de que la utilización, divulgación o copia 
sin autorización está prohibida en virtud de la legislación vigente. 
Si ha recibido este mensaje por error, se ruega lo comunique 
inmediatamente por esta misma vía y proceda a su destrucción.

This message is intended exclusively for its addressee and may contain 
information that is CONFIDENTIAL and protected by professional 
privilege. If you are not the intended recipient you are hereby notified 
that any dissemination, copy or disclosure of this communication is 
strictly prohibited by law. If this message has been received in error, 
please immediately notify us via e-mail.

**Reply by maximumspatium** [reply]
You're welcome!

Yes, the need to keep different Java versions to be able to run different Java programs is a very old and widespread bug known as the classpath bug. There is nothing we can do about that.

IIRC, the most recent code resided in the development branch requires Java 11. You can change to the development branch using this commands:
```
git fetch
git checkout development
```

Then simply rebuild Audiveris as described above (`./gradlew clean build`)
